### PR TITLE
refactor(gui-v2): new layout

### PR DESCRIPTION
**Pull request by pranavxc**
Signed-off-by: Pranav C <pranavxc@gmail.com>

## Change Summary

Provide summary of changes with issue number if any.

## Change type

- [ ] feat: (new feature for the user, not a new feature for build script)
- [ ] fix: (bug fix for the user, not a fix to a build script)
- [ ] docs: (changes to the documentation)
- [ ] style: (formatting, missing semi colons, etc; no production code change)
- [x] refactor: (refactoring production code, eg. renaming a variable)
- [ ] test: (adding missing tests, refactoring tests; no production code change)
- [ ] chore: (updating grunt tasks etc; no production code change)

## Test/ Verification

Provide summary of changes.

## Additional information / screenshots (optional)

Anything for maintainers to be made aware of

### CODE REVIEW

File: packages/nc-gui-v2/components/dashboard/TreeView.vue [hunk 1]
```diff
@@ -273,10 +370,15 @@ const activeTable = computed(() => {
...
+     Todo : move the component
+     <GithubStarButton />
+     -->
 
-    <div class="items-center flex justify-center mb-1">
-      <GithubStarButton />
+      <GeneralShareBaseButton class="!mr-0" />
     </div>
 
     <DlgTableCreate v-if="tableCreateDlg" v-model="tableCreateDlg" />
```
Comment (bcakmakoglu) [inline packages/nc-gui-v2/components/dashboard/TreeView.vue:384]: Replace this with

```js
function openTableCreateDialog() {
  $e('a:actions:create-table')

  const isOpen = ref(true)

  const { close } = useDialog(DlgTableCreate, {
    'modelValue': isOpen,
    'onUpdate:modelValue': closeDialog,
  })

  function closeDialog() {
    isOpen.value = false

    close(1000)
  }
}
```

File: packages/nc-gui-v2/components/general/Language.vue [hunk 2]
```diff
@@ -37,7 +39,36 @@ onMounted(() => {
...
+        <div
+          :class="lang === locale ? '!font-semibold !text-primary' : ''"
+          class="nc-project-menu-item capitalize group-hover:text-pink-500"
+        >
+          {{ Language[lang] || lang }}
+        </div>
+      </a-menu-item>
+
+      <a-menu-item>
+        <a
```
Comment (bcakmakoglu) [inline packages/nc-gui-v2/components/general/Language.vue:61]: I think it'll be better to just split the language component into two parts, one just contains the menu the other icon + menu - that way we avoid code duplication it stays more readable :)

File: packages/nc-gui-v2/components/smartsheet/sidebar/toolbar/ToggleDrawer.vue [hunk 3]
```diff
@@ -0,0 +1,18 @@
+<script setup lang="ts">
+/** Sidebar visible */
+const { isOpen, toggle } = useSidebar({ storageKey: 'nc-right-sidebar' })
+</script>
+
+<template>
+  <a-tooltip :placement="isOpen ? 'bottomRight' : 'left'">
```
Comment (bcakmakoglu) [inline packages/nc-gui-v2/components/smartsheet/sidebar/toolbar/ToggleDrawer.vue:7]: Can we put a debounce on the tooltip? like :mouse-enter-delay="0.5" or sth, so it doesn't instantly pop up?

**Review by bcakmakoglu — CHANGES_REQUESTED** [review]

File: packages/nc-gui-v2/components/general/Language.vue (hunk 2 above)
Comment (pranavxc) [reply packages/nc-gui-v2/components/general/Language.vue:61]: Sounds good will update into multiple component

**Pull request merged**